src/charging-station/ChargingStation.ts

```diff
@@ -672,6 +668,45 @@ export default class ChargingStation {
     this.messageBuffer.add(message);
   }
 
+  private buildStartMessage(): Record<string, unknown> {
+    return {
+      id: ChargingStationWorkerMessageEvents.STARTED,
+      payload: this.buildMessagePayload(),
+    };
+  }
+
+  private buildUpdateMessage() {
+    return {
+      id: ChargingStationWorkerMessageEvents.UPDATE,
+      payload: this.buildMessagePayload(),
+    };
+  }
+
+  private buildMessagePayload(): SimulatorUI {
+    return {
+      hashId: this.hashId,
+      stationInfo: this.buildPayloadStationInfo(),
+      connectors: Array.from(this.connectors.values()),
+    };
+  }
+
+  private buildPayloadStationInfo(): ChargingStationInfoUI {
+    return <ChargingStationInfoUI>{
+      chargingStationId: this.stationInfo.chargingStationId,
+      chargePointModel: this.stationInfo.chargePointModel,
+      chargePointVendor: this.stationInfo.chargePointVendor,
+      firmwareVersion: this.stationInfo.firmwareVersion,
+      numberOfConnectors: this.stationInfo.numberOfConnectors,
+    };
+  }
+
+  private buildStopMessage(): Record<string, unknown> {
+    return {
+      id: ChargingStationWorkerMessageEvents.STOPPED,
+      payload: { hashId: this.hashId },
+    };
+  }
```

Comment:
These are helpers to structure the internal messaging between main/workers. They should  be put into their own utils class as static methods to avoid consume memory just for building object literals.